Test/GenerationTester.hpp

```diff
@@ -87,60 +97,70 @@ class TestCase
   }                                                                                                           \
 
   BUILDER_METHOD(Clause*, input)
+  BUILDER_METHOD(Stack<Clause*>, context)
   BUILDER_METHOD(Stack<ClausePattern>, expected)
   BUILDER_METHOD(bool, premiseRedundant)
   BUILDER_METHOD(SimplifyingGeneratingInference*, rule)
+  BUILDER_METHOD(Stack<Indexing::Index*>, indices)
+  BUILDER_METHOD(OptionMap, options)
```

Comment:
I guess you mean we could instead pass all necessary options through ctors into `Induction`. It can be done, although it would be tedious to use due to the increasing number of options for induction. Maybe separating all induction options into a separate object inside `Options` would be the nicest way and we could pass this single object through ctors.

I don't know if it aligns with the effort to make the current options more hierarchical though. Also, it would increase the size of change in this PR significantly. What do you think?

Reply:
Yess i kind of thought that might be the case. 
If it's quite some work, don't mind. 
I think we'd kind of need a bigger refactor to untangle the options code from other parts of the code base..

Reply:
I ended up using `getOptions()` in `Induction` from `InferenceEngine` which doesn't use the global options. But I couldn't eliminate modifying `env.options` from `GenerationTester` due to a massive amount of static calls in `InductionHelper` which is used by term indices as well, and would ultimately require supplying the test indices themselves with the options from the test... At this point the modifications where so huge that I guess a separate PR would be much cleaner.

Reply:
Fair enough! Looks all neat and good to go!